include/aspect/world_builder/continental_plate.h

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_initial_temperature_world_generator_modules_constant_h
+#define _aspect_initial_temperature_world_generator_modules_constant_h
+
+#include <aspect/world_builder/interface.h>
+
+namespace aspect
+{
+  namespace WorldBuilder
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements zero initial topography.
+     *
+     * @ingroup InitialTopographyModels
+     */
+    template <int dim>
+    class Constant : public Interface<dim>
+    {
+      public:
+
+        /**
+         * Return the temperature of an object.
+         */
+        virtual
+        void temperature (const Point<dim> &position,
+                          const unsigned int object_number,
+                          const WorldBuilderParameters<dim> &world_generator_parameters,
```

Comment:
Hmm, I do not like this very much. Which `WorldBuilderParameters` might be needed by this plugin? Should not all things either be World builder parameter (in which case World Builder handles whatever should happen with them), or local parameters of this plugin?

Reply:
I personally prefer to hand over the previous temperature as `const double temperature` and return the new temperature as `double`. But that might be a personal preference.

Reply:
Yes, this had to do with the old structure, I still have to rethink the approach. I only added this file for now to show that the plugin structure with the declaration works.